src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -803,6 +804,409 @@ class CreateIndexStatementSegment(BaseSegment):
     )
 
 
+class AlterIndexStatementSegment(BaseSegment):
+    """An ALTER INDEX statement.
+
+    As per.
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-index-transact-sql?view=sql-server-ver15
+    """
+
+    type = "alter_index_statement"
+
+    _low_priority_lock_wait = Sequence(
+        "WAIT_AT_LOW_PRIORITY",
+        Bracketed(
+            Delimited(
+                Sequence(
+                    "MAX_DURATION",
+                    Ref("EqualsSegment"),
+                    Ref("NumericLiteralSegment"),
+                    Ref.keyword("MINUTES", optional=True),
+                ),
+                Sequence(
+                    "ABORT_AFTER_WAIT",
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        "NONE",
+                        "SELF",
+                        "BLOCKERS",
+                    ),
+                ),
+            ),
+        ),
+    )
+
+    _on_partitions = Sequence(
+        Sequence(
+            "ON",
+            "PARTITIONS",
+        ),
+        Bracketed(
+            Delimited(
+                Ref("NumericLiteralSegment"),
+            ),
+            Sequence(
+                "TO",
+                Ref("NumericLiteralSegment"),
+                optional=True,
+            ),
+        ),
+        optional=True,
+    )
+
+    _rebuild_index_option = AnyNumberOf(
+        Sequence(
+            "PAD_INDEX",
+            Ref("EqualsSegment"),
+            OneOf(
+                "ON",
+                "OFF",
+            ),
+        ),
+        Sequence(
+            "FILLFACTOR",
+            Ref("EqualsSegment"),
+            Ref("NumericLiteralSegment"),
+        ),
+        Sequence(
```

Comment:
Same feedback here as on the other PR, if you can bundle into a OneOf() where the sequence is cleaner, it reads a lot easier